### PR TITLE
Add Amusement Parks Index and Details Page

### PR DESCRIPTION
**Pull request by nickzahlis**
In this PR, we add the Index and Details pages for amusement parks as well as tests for both.

### CODE REVIEW

File: frontend/src/tests/pages/AmusementParks/AmusementParksDetailsPage.test.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+jest.mock('main/utils/amusementParksUtils', () => {
+    return {
+        __esModule: true,
+        amusementParkUtils: {
+            getById: (_id) => {
+                return {
+                    amusementParks: {
+                        id: 3,
+                        name: "Disneyland Park",
+                        description: "Disney-themed amusement park rides"
```
Comment (wq12li) [inline frontend/src/tests/pages/AmusementParks/AmusementParksDetailsPage.test.js:24]: Could use the "address" field here.

File: frontend/src/tests/pages/AmusementParks/AmusementParksDetailsPage.test.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    test("loads the correct fields, and no buttons", async () => {
+        render(
+            <QueryClientProvider client={queryClient}>
+                <MemoryRouter>
+                    <AmusementParksDetailsPage />
+                </MemoryRouter>
+            </QueryClientProvider>
+        );
+        expect(screen.getByText("Disneyland Park")).toBeInTheDocument();
+        expect(screen.getByText("Disney-themed amusement park rides")).toBeInTheDocument();
```
Comment (wq12li) [inline frontend/src/tests/pages/AmusementParks/AmusementParksDetailsPage.test.js:54]: Test for address

File: frontend/src/tests/pages/AmusementParks/AmusementParksIndexPage.test.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            </QueryClientProvider>
+        );
+
+        const createAmusementParksButton = screen.getByText("Create AmusementParks");
+        expect(createAmusementParksButton).toBeInTheDocument();
+        expect(createAmusementParksButton).toHaveAttribute("style", "float: right;");
+
+        const name = screen.getByText("Disneyland Park");
+        expect(name).toBeInTheDocument();
+
```
Comment (wq12li) [inline frontend/src/tests/pages/AmusementParks/AmusementParksIndexPage.test.js:67]: Test for address

File: frontend/src/tests/pages/AmusementParks/AmusementParksIndexPage.test.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            <QueryClientProvider client={queryClient}>
+                <MemoryRouter>
+                    <AmusementParksIndexPage />
+                </MemoryRouter>
+            </QueryClientProvider>
+        );
+
+        const name = screen.getByText("Disneyland Park");
+        expect(name).toBeInTheDocument();
+
```
Comment (wq12li) [inline frontend/src/tests/pages/AmusementParks/AmusementParksIndexPage.test.js:90]: address

File: frontend/src/tests/pages/AmusementParks/AmusementParksIndexPage.test.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        expect(mockDelete).toHaveBeenCalledTimes(1);
+        expect(mockDelete).toHaveBeenCalledWith(3);
+
+        await waitFor(() => expect(mockNavigate).toHaveBeenCalledWith("/amusementParks"));
+
+
+        // assert - check that the console.log was called with the expected message
+        expect(console.log).toHaveBeenCalled();
+        const message = console.log.mock.calls[0][0];
+        const expectedMessage = `AmusementParksIndexPage deleteCallback: {"id":3,"name":"Disneyland Park","description":"Disney-themed amusement park rides"}`;
```
Comment (wq12li) [inline frontend/src/tests/pages/AmusementParks/AmusementParksIndexPage.test.js:108]: address

**Comment by nickzahlis** [comment]
closed pr to fix tests

**Pull request closed without merging**